### PR TITLE
✨ Added warning when pushing sensitive files

### PR DESCRIPTION
**Pull request by Mario-SO**
As discussed in [here](https://github.com/mario-eth/soldeer/issues/67#issuecomment-2192539104) for issue #67, added a user warning when pushing .env or .git/

PS: Missing tests, looking forward to feedback.

### CODE REVIEW

**Comment by mario-eth** [comment]
It seems that we differ in fmt settings? cause i see there are many changes on how the imports are done?
i use the foundry ones:
https://github.com/foundry-rs/foundry/blob/master/CONTRIBUTING.md

File: src/lib.rs [hunk 1]
```diff
@@ -271,6 +251,19 @@ pub async fn run(command: Subcommands) -> Result<(), SoldeerError> {
             }
         }
         Subcommands::Push(push) => {
+            let path = push
+                .path
+                .unwrap_or(get_current_working_dir().to_str().unwrap().to_string());
+            let path_buf = PathBuf::from(&path);
+
+            // Check for sensitive files or directories
+            if utils::check_for_sensitive_files_or_directories_recursive(&path_buf) {
```
Comment (mario-eth) [inline src/lib.rs:260]: just import this so you can use it `check_for_sensitive_files_or_directories_recursive`

File: src/lib.rs [hunk 2]
```diff
@@ -637,4 +612,49 @@ libs = ["dependencies"]
...
+
+        // Check if the .env file exists
+        assert!(env_file_path.exists());
+
+        // Clean up
+        let _ = remove_file(&env_file_path);
+        let _ = remove_dir_all(&test_dir);
+
+        // Restore the original prompt function
+        // utils::prompt_user_for_confirmation = original_prompt;
```
Comment (mario-eth) [inline src/lib.rs:658]: is this needed?

File: src/utils.rs [hunk 3]
```diff
@@ -127,3 +117,47 @@ pub fn get_base_url() -> String {
         "https://api.soldeer.xyz".to_string()
     }
 }
+
+// Function to check for the presence of sensitive files or directories
+pub fn check_for_sensitive_files_or_directories(path: &Path) -> bool {
+    let env_file_exists = path.join(".env").exists();
+    let git_dir_exists = path.join(".git").exists();
```
Comment (mario-eth) [inline src/utils.rs:124]: maybe we can just look for any `.file` we have? and just display them all? 
And let the user decide if it's safe?

Comment (mario-eth) [reply src/utils.rs:124]: cause it may be more `.` files

Comment (Mario-SO) [reply src/utils.rs:124]: Makes sense yeah.

**Comment by Mario-SO** [comment]
> It seems that we differ in fmt settings? cause i see there are many changes on how the imports are done?
> 
> i use the foundry ones:
> 
> https://github.com/foundry-rs/foundry/blob/master/CONTRIBUTING.md

Crap, you are totally right, my bad. Will fix and PR again.

PS: Also saw the rest of the comments you made.

File: src/lib.rs (hunk 1 above)
Comment (Mario-SO) [reply src/lib.rs:260]: I don't get what you mean here

File: src/lib.rs [hunk 4]
```diff
@@ -271,6 +269,19 @@ pub async fn run(command: Subcommands) -> Result<(), SoldeerError> {
             }
         }
         Subcommands::Push(push) => {
+            let path = push
+                .path
+                .unwrap_or(get_current_working_dir().to_str().unwrap().to_string());
+            let path_buf = PathBuf::from(&path);
+
+            // Check for sensitive files or directories
+            if utils::check_for_sensitive_files_or_directories_recursive(&path_buf) {
```
Comment (mario-eth) [inline src/lib.rs:278]:
```suggestion
            if check_for_sensitive_files_or_directories_recursive(&path_buf) {
```

Sorry i just meant like this.

Comment (mario-eth) [reply src/lib.rs:278]: also the name is a bit tbh, can we work on something smaller?

Comment (Mario-SO) [reply src/lib.rs:278]: Oh ok I get you now, also, yeah the name is way too descriptive XD

**Comment by mario-eth** [comment]
you have to run 
`cargo +nightly clippy --workspace --all-targets --all-features`
And resolve them

and 
`cargo +nightly fmt --all --check`

Cause the CI fails

**Comment by Mario-SO** [comment]
Ok I see, sorry for all the inconveniences this PR is causing first of all.

Also wanted to ask, when running `cargo +nightly clippy --workspace --all-targets --all-features`, there is a warning in file `dependency_downloader.rs` which is a file I didn't touch at all. The warning is this one

```rust
warning: this function depends on never type fallback being `()`
  --> src/dependency_downloader.rs:18:1
   |
18 | / pub async fn download_dependencies(
19 | |     dependencies: &[Dependency],
20 | |     clean: bool,
21 | | ) -> Result<(), DownloadError> {
   | |______________________________^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
   = help: specify the types explicitly
note: in edition 2024, the requirement `!: std::iter::FromIterator<()>` will fail
  --> src/dependency_downloader.rs:35:16
   |
35 |     .collect::<Result<_, _>>()?;
   |                ^^^^^^^^^^^^
   = note: `#[warn(dependency_on_unit_never_type_fallback)]` on by default
```

The rest of the warnings are resolved and also ran the `cargo +nightly fmt --all --check`.

I will push again and await your feedback once more.

Again sorry to bother this much with such a minor PR.

**Comment by mario-eth** [comment]
> Ok I see, sorry for all the inconveniences this PR is causing first of all.
> 
> Also wanted to ask, when running `cargo +nightly clippy --workspace --all-targets --all-features`, there is a warning in file `dependency_downloader.rs` which is a file I didn't touch at all. The warning is this one
> 
> ```rust
> warning: this function depends on never type fallback being `()`
>   --> src/dependency_downloader.rs:18:1
>    |
> 18 | / pub async fn download_dependencies(
> 19 | |     dependencies: &[Dependency],
> 20 | |     clean: bool,
> 21 | | ) -> Result<(), DownloadError> {
>    | |______________________________^
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
>    = help: specify the types explicitly
> note: in edition 2024, the requirement `!: std::iter::FromIterator<()>` will fail
>   --> src/dependency_downloader.rs:35:16
>    |
> 35 |     .collect::<Result<_, _>>()?;
>    |                ^^^^^^^^^^^^
>    = note: `#[warn(dependency_on_unit_never_type_fallback)]` on by default
> ```
> 
> The rest of the warnings are resolved and also ran the `cargo +nightly fmt --all --check`.
> 
> I will push again and await your feedback once more.
> 
> Again sorry to bother this much with such a minor PR.

ah no worries at all, thank you for taking the time to get this done.
That is weird, it seems it's like a new compiler warning, can you solve it?

**Comment by Mario-SO** [comment]
Sure, will solve it asap

**Comment by mario-eth** [comment]
actually i m gonna solve them cause it seems there are more now after I did an update. will solve then if everything is ok I m gonna merge this one

**Comment by Mario-SO** [comment]
It no longer pops any warnings.

**Pull request merged**